exchanges/stream/stream_match.go

```diff
@@ -5,76 +5,60 @@ import (
 	"sync"
 )
 
+var errSignatureCollision = errors.New("signature collision")
+
 // NewMatch returns a new Match
 func NewMatch() *Match {
-	return &Match{
-		m: make(map[interface{}]chan []byte),
-	}
+	return &Match{m: make(map[any]chan<- []byte)}
 }
 
 // Match is a distributed subtype that handles the matching of requests and
 // responses in a timely manner, reducing the need to differentiate between
 // connections. Stream systems fan in all incoming payloads to one routine for
 // processing.
 type Match struct {
-	m  map[interface{}]chan []byte
+	m  map[any]chan<- []byte
 	mu sync.Mutex
 }
 
-// Matcher defines a payload matching return mechanism
-type Matcher struct {
-	C   chan []byte
-	sig interface{}
-	m   *Match
-}
-
 // Incoming matches with request, disregarding the returned payload
-func (m *Match) Incoming(signature interface{}) bool {
+func (m *Match) Incoming(signature any) bool {
 	return m.IncomingWithData(signature, nil)
 }
```

Comment:
I think we should remove Incoming.
It's only used in tests.
You can always move it to test pkg I think.